lab_6_pipeline/pipeline.py

```diff
@@ -27,16 +53,46 @@ def __init__(self, path_to_raw_txt_data: pathlib.Path) -> None:
         Args:
             path_to_raw_txt_data (pathlib.Path): Path to raw txt data
         """
+        self.path_to_raw_txt_data = path_to_raw_txt_data
+        self._storage = {}
+        self._validate_dataset()
+        self._scan_dataset()
 
     def _validate_dataset(self) -> None:
         """
         Validate folder with assets.
         """
+        if not self.path_to_raw_txt_data.exists():
+            raise FileNotFoundError
+
+        if not self.path_to_raw_txt_data.is_dir():
+            raise NotADirectoryError
+
+        if not list(self.path_to_raw_txt_data.iterdir()):
+            raise EmptyDirectoryError
+
+        raw_files = list(self.path_to_raw_txt_data.glob('*_raw.txt'))
+        meta_files = list(self.path_to_raw_txt_data.glob('*_meta.json'))
+        if len(raw_files) != len(meta_files):
+            raise InconsistentDatasetError
+
+        sorted_raw_files = sorted(raw_files, key=lambda x: get_article_id_from_filepath(x))
+        sorted_meta_files = sorted(meta_files, key=lambda x: get_article_id_from_filepath(x))
+
+        for index, (raw_file, meta_file) in enumerate(zip(sorted_raw_files, sorted_meta_files)):
+            if (index + 1 != get_article_id_from_filepath(raw_file)
```

Comment:
Can we avoid adding 1 at each step by directly setting `enumerate` to start from 1?

Reply:
Fixed